src/org/traccar/database/ComputedAttributesManager.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.database;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.traccar.Context;
+import org.traccar.helper.Log;
+import org.traccar.model.ComputedAttribute;
+
+public class ComputedAttributesManager {
+
+    private final DataManager dataManager;
+
+    private final Map<Long, Set<ComputedAttribute>> deviceComputedAttributes = new ConcurrentHashMap<>();
```

Comment:
I think we need to think of a way of applying attributes for all devices and/or specific protocol and/or groups. I think in most cases users would have some number of devices of the same model and similar configuration.

Probably applying to a group would be the most universal solution, but then it might interfere with some other grouping that user might want to do.

What do you think?